src/modules/fancyzones/lib/ZoneSet.cpp

```diff
@@ -13,86 +14,32 @@ namespace
-            .rowsPercents = { 10000 },
-            .columnsPercents = { 10000 },
-            .cellChildMap = { { 0 } } },
-        /* 2 */
-        {
-            .rows = 1,
-            .columns = 2,
-            .rowsPercents = { 10000 },
-            .columnsPercents = { 6667, 3333 },
-            .cellChildMap = { { 0, 1 } } },
-        /* 3 */
-        {
-            .rows = 1,
-            .columns = 3,
-            .rowsPercents = { 10000 },
-            .columnsPercents = { 2500, 5000, 2500 },
-            .cellChildMap = { { 0, 1, 2 } } },
-        /* 4 */
-        {
-            .rows = 2,
-            .columns = 3,
-            .rowsPercents = { 5000, 5000 },
-            .columnsPercents = { 2500, 5000, 2500 },
-            .cellChildMap = { { 0, 1, 2 }, { 0, 1, 3 } } },
-        /* 5 */
-        {
-            .rows = 2,
-            .columns = 3,
-            .rowsPercents = { 5000, 5000 },
-            .columnsPercents = { 2500, 5000, 2500 },
-            .cellChildMap = { { 0, 1, 2 }, { 3, 1, 4 } } },
-        /* 6 */
-        {
-            .rows = 3,
-            .columns = 3,
-            .rowsPercents = { 3333, 3334, 3333 },
-            .columnsPercents = { 2500, 5000, 2500 },
-            .cellChildMap = { { 0, 1, 2 }, { 0, 1, 3 }, { 4, 1, 5 } } },
-        /* 7 */
-        {
-            .rows = 3,
-            .columns = 3,
-            .rowsPercents = { 3333, 3334, 3333 },
-            .columnsPercents = { 2500, 5000, 2500 },
-            .cellChildMap = { { 0, 1, 2 }, { 3, 1, 4 }, { 5, 1, 6 } } },
-        /* 8 */
-        {
-            .rows = 3,
-            .columns = 4,
-            .rowsPercents = { 3333, 3334, 3333 },
-            .columnsPercents = { 2500, 2500, 2500, 2500 },
-            .cellChildMap = { { 0, 1, 2, 3 }, { 4, 1, 2, 5 }, { 6, 1, 2, 7 } } },
-        /* 9 */
-        {
-            .rows = 3,
-            .columns = 4,
-            .rowsPercents = { 3333, 3334, 3333 },
-            .columnsPercents = { 2500, 2500, 2500, 2500 },
-            .cellChildMap = { { 0, 1, 2, 3 }, { 4, 1, 2, 5 }, { 6, 1, 7, 8 } } },
-        /* 10 */
-        {
-            .rows = 3,
-            .columns = 4,
-            .rowsPercents = { 3333, 3334, 3333 },
-            .columnsPercents = { 2500, 2500, 2500, 2500 },
-            .cellChildMap = { { 0, 1, 2, 3 }, { 4, 1, 5, 6 }, { 7, 1, 8, 9 } } },
-        /* 11 */
-        {
-            .rows = 3,
-            .columns = 4,
-            .rowsPercents = { 3333, 3334, 3333 },
-            .columnsPercents = { 2500, 2500, 2500, 2500 },
-            .cellChildMap = { { 0, 1, 2, 3 }, { 4, 1, 5, 6 }, { 7, 8, 9, 10 } } }
-    };
+    auto l = JSONHelpers::GridLayoutInfo(1, 1);
+        // PriorityGrid layout is unique for zoneCount <= 11. For zoneCount > 11 PriorityGrid is same as Grid
+        JSONHelpers::GridLayoutInfo predefinedPriorityGridLayouts[11] = {
+            /* 1 */
+            JSONHelpers::GridLayoutInfo(1, 1, { 10000 }, { 10000 }, { { 0 } }),
```

Comment:
I would prefer if you keep the original code since clarity is always to be preferred to compactness.

Reply:
I agree

Reply:
Seraphima explained to me the reason of the change, if we can add a label to the arguments, so it's clear what arguments are we passing, that's also fine with me.